util/design/gen-flash-img.py

```diff
@@ -132,40 +130,61 @@ def _get_flash_scrambling_configs(otp_vmem_file: str,
     try:
         otp_vmem = Path(otp_vmem_file).read_text()
     except IOError:
-        raise Exception(f"Unable to open {otp_vmem}")
+        raise Exception(f"Unable to open {otp_vmem_file}")
     otp_vmem_lines = re.findall(r"^@.*$", otp_vmem, flags=re.MULTILINE)
 
-    # TODO: Retrieve partition with the flash scrambling enablement flags.
-    configs.scrambling_enabled = False
-
-    # Retrieve SECRET1 partition which contains the flash scrambling key seeds,
-    # stripping ECC bits from each data word when processing.
-    data_blocks_64bit = []
-    data_block_64bit = 0
+    # Retrieve OTP data from the following partitions:
+    # - CREATOR_SW_CFG: which contains the flash scramble enablement flag, and
+    # - SECRET1: which contains the flash scrambling key seeds.
+    # Note, we strip ECC bits from each data word when processing.
+    flash_data_default_cfg = None
+    secret1_data_blocks = []
+    otp_data_block = 0
     idx = 0
     for line in otp_vmem_lines:
-        if OTP_SECRET1_RE.search(line):
+        if (OTP_FLASH_DATA_DEFAULT_CFG_RE.search(line) or
+                OTP_SECRET1_RE.search(line)):
             otp_data_word_w_ecc = int(line.split()[1], OTP_WORD_SIZE)
             otp_data_word = otp_data_word_w_ecc & (2**OTP_WORD_SIZE - 1)
-            data_block_64bit |= otp_data_word << (idx * OTP_WORD_SIZE)
+            otp_data_block |= otp_data_word << (idx * OTP_WORD_SIZE)
             idx += 1
-            if idx == (64 // OTP_WORD_SIZE):
-                data_blocks_64bit.append(data_block_64bit)
-                data_block_64bit = 0
-                idx = 0
+            if OTP_FLASH_DATA_DEFAULT_CFG_RE.search(line):
+                if idx == (OTP_FLASH_DATA_DEFAULT_CFG_BLOCK_SIZE //
+                           OTP_WORD_SIZE):
+                    flash_data_default_cfg = otp_data_block & 0xff
+                    # If flash data scrambling is disabled, then we can return
+                    # early to save execution time.
+                    if flash_data_default_cfg != 0x6:
```

Comment:
Probably should use something like `MUBI4_TRUE` here.

Reply:
fixed.